app/routes/languages.rb

```diff
@@ -41,6 +44,23 @@ class Languages < Core
         end
       end
 
+      get '/languages/:track_id/:topic' do |track_id, topic|
+        _, body = X::Xapi.get('tracks', track_id)
+        parsed_body = JSON.parse(body)
+        if parsed_body['error'] == "No track '#{track_id}'"
+          erb :"languages/not_found", locals: { track_id: track_id }
+        elsif !TOPICS.include?(topic.to_sym)
```

Comment:
What do you think of checking for the topics at the beginning of the method? This way if we have an invalid topic we don't make a request for the Xapi.